components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/IdentityManagementEndpointUtil.java

```diff
@@ -188,11 +189,17 @@ public static final String getUserPortalUrl(String userPortalUrl, String tenantD
         try {
             if (StringUtils.isNotEmpty(tenantDomain)) {
                 ApplicationDataRetrievalClient applicationDataRetrievalClient = new ApplicationDataRetrievalClient();
+                String myAccountAccessUrl;
                 try {
-                    String myAccountAccessUrl = applicationDataRetrievalClient.getApplicationAccessURL(SUPER_TENANT,
-                            My_ACCOUNT_APPLICATION_NAME);
+                    if (CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME == true) {
+                        myAccountAccessUrl = applicationDataRetrievalClient.getApplicationAccessURL(SUPER_TENANT,
+                                My_ACCOUNT_APPLICATION_NAME);
+                    } else {
+                        myAccountAccessUrl = applicationDataRetrievalClient.getApplicationAccessURL(tenantDomain,
+                                My_ACCOUNT_APPLICATION_NAME);
+                    }
                     if (StringUtils.isNotEmpty(myAccountAccessUrl)) {
-                        return replaceUserTenantHintPlaceholder(myAccountAccessUrl, tenantDomain);
```

Comment:
Don't we need to do this when `CarbonConstants.ENABLE_LEGACY_AUTHZ_RUNTIME == true`?

Reply:
If legacy runtime is enabled, we will be using the SAAS app. So the method to log into a tenanted user in the SAAS app is to provide the username as `name@tenantDomain`. So, it is not required.

Reply:
But if the url contains the `USER_TENANT_HINT_PLACE_HOLDER` placeholder by any change, don't we need to replace with proper values.

Reply:
As I tested, `USER_TENANT_HINT_PLACE_HOLDER` was not present when we retrieve the super tenant myaccount access url in legacy mode. 

And in default runtime, we directly retrieve the access url for the specific tenant. Hence, this is not required in either.